oauth_provider/README.rst

```diff
@@ -0,0 +1,421 @@
+
+Once configured, you must give these information to your client application :
+
+#. Client identifier : Identifies the application (to be able to check allowed scopes and redirect URIs)
+#. Allowed scopes : The codes of scopes allowed for this client
+#. URLs for the requests :
+
+  - Authorization request : http://odoo.example.com/oauth2/authorize
+  - Token request : http://odoo.example.com/oauth2/token
+  - Token information request : http://odoo.example.com/oauth2/tokeninfo
+     Parameters : access_token
+  - User information request : http://odoo.example.com/oauth2/userinfo
+     Parameters : access_token
+  - Any other model information request (depending on the scopes) : http://odoo.example.com/oauth2/data
+     Parameters : access_token and model
+
+For example, to configure the *auth_oauth* Odoo module as a client, you will enter these values :
+
+- Provider name : Anything you want
+- Client ID : The identifier of the client configured in your Odoo Provider instance
+- Body : Text displayed on Odoo's login page link
+- Authentication URL : http://odoo.example.com/oauth2/authorize
+- Scope : A space separated list of scope codes allowed to the client in your Odoo Provider instance
+- Validation URL : http://odoo.example.com/oauth2/tokeninfo
+- Data URL : http://odoo.example.com/oauth2/userinfo
+
+Following is an example of obtaining an OAuth2 token in Python using the Legacy Application type (requires `requests_oauth`):
+
+.. code-block:: python
+
+   import os
+   import requests
+
+   from oauthlib.oauth2 import LegacyApplicationClient
+   from requests_oauthlib import OAuth2Session
+
+   # Allows for no HTTPS verification. DO NOT USE IN PRODUCTION!
+   os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
+
+   client_id = '2ec860db-ce81-49c2-b4b7-c71bcbee481f'
+   client_secret = 'Test'
+   username = 'admin'
+   password = 'admin'
+
+   oauth = OAuth2Session(client=LegacyApplicationClient(client_id=client_id))
+   token = oauth.fetch_token(
+       token_url='http://localhost:8060/oauth2/token',
+       username=username,
+       password=password,
+       client_id=client_id,
+       client_secret=client_secret,
+       scope=['public-pricelist', 'pricelist-item', 'medical-medicament'],
+   )
+
+REST API
+========
+
+This module also includes a basic REST-style CRUD interface backed by OAuth2 authentication.
+
+List
+----
+
+Return allowed information from all records, with optional query.
+
+Route: `/api/rest/1.0/<string:model>`
+
+Accepts: `GET`
+
+Args:
+   access_token (str): OAuth2 access token to utilize for the
+       operation.
+   model (str): Name of model to operate on.
+   domain (array, optional): Domain to apply to the search, in the
+       standard Odoo format. This will be appended to the scope's
+       pre-existing filter.
+
+Returns:
+   array of objs: List of data mappings matching query.
+
+Example:
+
+Bash:
+
+.. code-block:: bash
+
+   curl -H "Content-Type: application/json" \
```

Comment:
NVM this was just someone that couldn't perform the OAuth dance correctly. Bearer token is not required past the OAuth authentication & token exchange phases.

That said, I think we're breaking RFC here:

>    The method in which the client utilizes the access token to
>   authenticate with the resource server depends on the type of access
>   token issued by the authorization server.  Typically, it involves
>   using the HTTP "Authorization" request header field [RFC2617] with an
>   authentication scheme defined by the specification of the access
>   token type used, such as [RFC6750].

So I think instead we need to scrape the token from the request headers for use, instead of requiring as call params. Thoughts?